### PR TITLE
feat(Scripts/Commands): Allow to pass player as argument in instance …

### PR DESCRIPTION
**Pull request by Kitzunu**
…unbind

* closes https://github.com/azerothcore/azerothcore-wotlk/issues/10516

## Changes Proposed:
-  Allow to pass player argument in 

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10516

## Tests Performed:

- tested ingame

## How to Test the Changes:

1. .instance unbind [player] <mapid|all> [difficulty]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
now you can even use it throught the console :D

File: src/server/scripts/Commands/cs_instance.cpp [hunk 1]
```diff
@@ -109,13 +116,13 @@ class instance_commandscript : public CommandScript
             for (BoundInstancesMap::const_iterator itr = m_boundInstances.begin(); itr != m_boundInstances.end();)
             {
                 InstanceSave const* save = itr->second.save;
-                if (itr->first != player->GetMapId() && (!mapId || mapId == itr->first) && (!difficultyArg || difficultyArg == save->GetDifficulty()))
+                if (itr->first != player->GetConnectedPlayer()->GetMapId() && (!mapId || mapId == itr->first) && (!difficultyArg || difficultyArg == save->GetDifficulty()))
```
Comment (Winfidonarleyan) [inline src/server/scripts/Commands/cs_instance.cpp:119]: if player offline - crash

Comment (Kitzunu) [reply src/server/scripts/Commands/cs_instance.cpp:119]: true... will add null check

**Comment by Winfidonarleyan** [comment]
@Kitzunu check `HandleReviveCommand` for example using `Optional<PlayerIdentifier>`

File: src/server/scripts/Commands/cs_instance.cpp [hunk 2]
```diff
@@ -87,11 +87,18 @@ class instance_commandscript : public CommandScript
...
     {
-        Player* player = handler->getSelectedPlayer();
+        // Character name must be provided when using this from console.
+        if (!player && !handler->GetSession())
+        {
+            handler->PSendSysMessage(LANG_CMD_SYNTAX);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
```
Comment (Winfidonarleyan) [inline src/server/scripts/Commands/cs_instance.cpp:99]:
```cpp
if (!target)
        {
            target = PlayerIdentifier::FromTargetOrSelf(handler);
        }

        if (!target || !target->IsConnected())
        {
            return false;
        }

        auto targetPlayer = target->GetConnectedPlayer();
```

**Comment by Maelthyrr** [comment]
@Kitzunu status? Need a test?

**Comment by Kitzunu** [comment]
abandoned, for now.

**Pull request closed without merging**